code/modules/power/supermatter/supermatter.dm

```diff
@@ -1,359 +1,580 @@
-	var/emergency_point = 400
+	var/warning_alert = "Danger! Crystal hyperstructure integrity faltering!"
+	var/damage_penalty_point = 550
+	var/emergency_point = 700
 	var/emergency_alert = "CRYSTAL DELAMINATION IMMINENT."
-	var/explosion_point = 600
+	var/explosion_point = 900
 
-	var/emergency_issued = 0
+	var/emergency_issued = FALSE
 
-	var/explosion_power = 8
+	var/explosion_power = 35
+	var/temp_factor = 30
 
 	var/lastwarning = 0				// Time in 1/10th of seconds since the last sent warning
-	var/last_zap = 0				// Time in 1/10th of seconds since the last tesla zap
 	var/power = 0
 
-	var/oxygen = 0					// Moving this up here for easier debugging.
+	var/n2comp = 0					// raw composition of each gas in the chamber, ranges from 0 to 1
+
+	var/plasmacomp = 0
+	var/o2comp = 0
+	var/co2comp = 0
+	var/n2ocomp = 0
+
+	var/combined_gas = 0
+	var/gasmix_power_ratio = 0
+	var/dynamic_heat_modifier = 1
+	var/dynamic_heat_resistance = 1
+	var/powerloss_inhibitor = 1
+	var/powerloss_dynamic_scaling= 0
+	var/power_transmission_bonus = 0
+	var/mole_heat_penalty = 0
+
+
+	var/matter_power = 0
 
 	//Temporary values so that we can optimize this
 	//How much the bullets damage should be multiplied by when it is added to the internal variables
 	var/config_bullet_energy = 2
+	//How much of the power is left after processing is finished?
+//	var/config_power_reduction_per_tick = 0.5
 	//How much hallucination should it produce per unit of power?
 	var/config_hallucination_power = 0.1
 
-	var/debug = 0
-
-	var/disable_adminwarn = FALSE
-
-	var/aw_normal = FALSE
-	var/aw_notify = FALSE
-	var/aw_warning = FALSE
-	var/aw_danger = FALSE
-	var/aw_emerg = FALSE
-	var/aw_delam = FALSE
-
 	var/obj/item/radio/radio
 
 	//for logging
-	var/has_been_powered = 0
-	var/has_reached_emergency = 0
+	var/has_been_powered = FALSE
+	var/has_reached_emergency = FALSE
 
-/obj/machinery/power/supermatter_shard/crystal
-	name = "supermatter crystal"
-	desc = "A strangely translucent and iridescent crystal."
-	base_icon_state = "darkmatter"
-	icon_state = "darkmatter"
-	anchored = TRUE
-	warning_point = 200
-	emergency_point = 2000
-	explosion_point = 3600
-	gasefficency = 0.25
-	explosion_power = 24
+	// For making hugbox supermatter
+	var/takes_damage = TRUE
+	var/produces_gas = TRUE
+	var/obj/effect/countdown/supermatter/countdown
 
+	var/is_main_engine = FALSE
 
-/obj/machinery/power/supermatter_shard/New()
+	var/datum/looping_sound/supermatter/soundloop
+
+	var/moveable = FALSE
+
+/obj/machinery/power/supermatter_crystal/Initialize(mapload)
 	. = ..()
-	GLOB.poi_list |= src
-	//Added to the atmos_machine process as the SM is highly coupled with the atmospherics system.
-	//Having the SM run at a different rate then atmospherics causes odd behavior.
 	SSair.atmos_machinery += src
+	countdown = new(src)
+	countdown.start()
+	GLOB.poi_list |= src
 	radio = new(src)
 	radio.listening = 0
+	radio.config(list("Engineering" = 0))
```

Comment:
Shouldnt this be this so it has the channel enabled?
```suggestion
	radio.config(list("Engineering" = 1))
```

Reply:
Most of the device that talk into a radio have these set to `0` for some reason (clonepod, sec doors, and the likes).

Reply:
Fair enough